lib/swagger-mock-validator/validate-spec-and-mock/validate-parsed-mock-response-body.ts

```diff
@@ -1,68 +1,119 @@
+  parsedMockInteraction: ParsedMockInteraction
+) => !parsedMockInteraction.responseBody.value;
 
 const isNotSupportedMediaType = (parsedSpecResponse: ParsedSpecResponse) =>
-    parsedSpecResponse.produces.value.length > 0 &&
-    !isMediaTypeSupported('application/json', parsedSpecResponse.produces.value);
+  parsedSpecResponse.produces.value.length > 0 &&
+  !isMediaTypeSupported('application/json', parsedSpecResponse.produces.value);
 
-const shouldSkipValidation = (parsedMockInteraction: ParsedMockInteraction, parsedSpecResponse: ParsedSpecResponse) =>
-    isMockInteractionWithoutResponseBody(parsedMockInteraction) || isNotSupportedMediaType(parsedSpecResponse);
+const shouldSkipValidation = (
+  parsedMockInteraction: ParsedMockInteraction,
+  parsedSpecResponse: ParsedSpecResponse
+) =>
+  isMockInteractionWithoutResponseBody(parsedMockInteraction) ||
+  isNotSupportedMediaType(parsedSpecResponse);
 
-export const validateParsedMockResponseBody = (parsedMockInteraction: ParsedMockInteraction,
-                                               parsedSpecResponse: ParsedSpecResponse) => {
-    if (shouldSkipValidation(parsedMockInteraction, parsedSpecResponse)) {
-        return [];
-    }
+export const validateParsedMockResponseBody = (
+  parsedMockInteraction: ParsedMockInteraction,
+  parsedSpecResponse: ParsedSpecResponse,
+  opts: Pick<
+    ValidateOptions,
+    'additionalPropertiesInResponse' | 'requiredPropertiesInResponse'
+  >
+) => {
+  if (shouldSkipValidation(parsedMockInteraction, parsedSpecResponse)) {
+    return [];
+  }
 
-    if (!parsedSpecResponse.schema) {
-        return [
-            result.build({
-                code: 'response.body.unknown',
-                message: 'No schema found for response body',
-                mockSegment: parsedMockInteraction.responseBody,
-                source: 'spec-mock-validation',
-                specSegment: parsedSpecResponse
-            })
-        ];
-    }
+  if (!parsedSpecResponse.schema) {
+    return [
+      result.build({
+        code: 'response.body.unknown',
+        message: 'No schema found for response body',
+        mockSegment: parsedMockInteraction.responseBody,
+        source: 'spec-mock-validation',
+        specSegment: parsedSpecResponse
+      })
+    ];
+  }
 
-    const responseBodyWithoutRequiredProperties = removeRequiredPropertiesFromSchema(parsedSpecResponse.schema);
+  let responseBodyToValidate: ParsedSpecJsonSchema = parsedSpecResponse.schema;
 
-    const validationErrors = validateJson(
-        responseBodyWithoutRequiredProperties,
-        parsedMockInteraction.responseBody.value
+  // tslint:disable:cyclomatic-complexity
+  if (!opts.additionalPropertiesInResponse) {
+    responseBodyToValidate = setAdditionalPropertiesToFalseInSchema(
+      parsedSpecResponse.schema
     );
+  }
+  if (!opts.requiredPropertiesInResponse) {
+    responseBodyToValidate = removeRequiredPropertiesFromSchema(
+      responseBodyToValidate
+    );
```

Comment:
Rather than overloading the method for removing retaining required properties, this allows for either flag to be set

original change was here https://bitbucket.org/rcrodrigueszup/swagger-pact-validator/commits/b13e08c30beb1c305c650786c89a44e62de0da2d?atlOrigin=eyJpIjoiMWRiZjlmZjhkYmE3NDg0Mzk3NWI3ODZhZjczNGQyODQiLCJwIjoiYmItY2hhdHMtaW50ZWdyYXRpb24ifQ